recipes/cpputest/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "verbose": [True, False],
+        "use_std_c_lib": ["ON", "OFF"],
+        "use_std_cpp_lib": ["ON", "OFF"],
+        "use_cpp11": ["ON", "OFF"],
+        "detect_mem_leaks": ["ON", "OFF"],
+        "extensions": ["ON", "OFF"],
+        "longlong": ["ON", "OFF"],
+        "coverage": ["ON", "OFF"],
+        "tests": ["ON", "OFF"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+        "verbose": False,
+        "use_std_c_lib": "ON",
+        "use_std_cpp_lib": "ON",
+        "use_cpp11": "ON",
+        "detect_mem_leaks": "ON",
+        "extensions": "ON",
+        "longlong": "ON",
+        "coverage": "OFF",
+        "tests": "OFF",
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        # tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+        #                       "set (CMAKE_POSITION_INDEPENDENT_CODE ON)", "")
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.verbose = self.options.verbose
+        self._cmake.definitions["STD_C"] = self.options.use_std_c_lib
+        self._cmake.definitions["STD_CPP"] = self.options.use_std_cpp_lib
+        self._cmake.definitions["C++11"] = self.options.use_cpp11
+        self._cmake.definitions["MEMORY_LEAK_DETECTION"] = self.options.detect_mem_leaks
+        self._cmake.definitions["EXTENSIONS"] = self.options.extensions
+        self._cmake.definitions["LONGLONG"] = self.options.longlong
+        self._cmake.definitions["COVERAGE"] = self.options.coverage
+        self._cmake.definitions["TESTS"] = self.options.tests
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "CppUTest", "cmake"))
+
+    def package_info(self):
```

Comment:
You need to explicitly set the pkg_config names

Reply:
I'm not exactly sure that I understand that, but I have improved code in this area in e8ee75a888ce472523cef58a88e35fb281be3bbe.

If that's not what you mean, please can I have a bit more info?

Reply:
You do not need file name ... the default is names

Reply:
```suggestion 
Self.cpp_info.names["pkg_config"] = "whatever the .pc file is that's removed"
```

Via mobile ⚠️